services/controllerhost/queueDepth.go

```diff
@@ -542,12 +549,27 @@ func (qdc *queueDepthCalculator) handleStartFrom(
 
 		// if doGaft == false, this just adjusts for retention, if applicable
 		if qualify {
-			trace += 100
 			consumerGroupExtent.WriteTime = common.Int64Ptr(int64(now))
-			consumerGroupExtent.AckLevelSeqNo = common.Int64Ptr(common.MaxInt64(
-				storeMetadata.beginSequence, // Retention may have removed some messages
-				int64(startFromSeq),         // Otherwise, act like we had just opened this extent at startFrom, i.e. don't count messages before startFrom
-			))
+
+			// retention may have purged some messages, account for that
+			if storeMetadata.beginSequence != math.MaxInt64 {
+
+				trace += 100
+
+				// the 'ack-level' corresponds to message that has already been read (ie 'acked');
+				// so reduce one from the 'beginSequence' to since that msg has not been read.
+				consumerGroupExtent.AckLevelSeqNo = common.Int64Ptr(common.MaxInt64(
+					storeMetadata.beginSequence-1, // Retention may have removed some messages
```

Comment:
No. Message seq-nums start from "1". If there are no messages in the extent, then the beginSequence would be "MaxInt64".